core/templates/components/summary-tile/topic-summary-tile.component.spec.ts

```diff
@@ -93,4 +96,17 @@ describe('TopicSummaryTileCompoennt', () => {
   it('should get darker thumbnail background color', () => {
     expect(component.getDarkerThumbnailBgColor()).toBe('#627876');
   });
+
+  it('should get correct translations', () => {
+    spyOn(i18nLanguageCodeService, 'getTopicTranslationKey').and.returnValue(
+      'I18N_TOPIC_12345axa_TITLE');
+    spyOn(i18nLanguageCodeService, 'hasTranslations').and.returnValue(
+      false);
+
+    component.ngOnInit();
+
+    expect(component.hasTranslationKey()).toBe(false);
+    expect(component.topicNameTranslationKey).toBe(
```

Comment:
I feel like I'm missing something here -- conceptually, shouldn't it be the value, not the key, that matches the topic summary name?

Reply:
so the the var `topicNameTranslationKey` stores either the name of the topic or the translation key based on if the language if english and the key has a translation.

Reply:
That is problematic. See comment above.

Reply:
Just noting that this comment thread hasn't been responded to yet.

Reply:
Have implemented it correctly